### DISCUSSION TITLE
Timestamp required for last content update of DevPortal in designer/editor mode (GUI and API-based)

### DISCUSSION THREAD

**Discussion by AnRei123** [Azure/api-management-developer-portal#2579] [Feature request]
##Problem statement

We currently do the editing of the DevPortal contents in a separate portal environment. From there, we deploy the updated contents to about 20 target environments. 
Before publishing the contents one of these target environments, it is important to know the timestamp when the content has been modified the last time to ensure that only controlled and released content is published.

Additionally, it is not possible to restrict the access to the Developer Portal in designer / editor mode separately. Everyone, who has access to the APIM portal is currently able to also modify the contents. Currently, there is no version control supported for content update of the DevPortal.

## Expected behavior

1. On the "Portal overview" page, in the APIM portal, on the "Revision" tab, a timestamp is missing to indicate when the content in the designer/editor mode has been modified the last time and a mapping if the latest content has been already published.
2. For each publishing revision, a modification timestamp for the published portal contents should be shown in addition. 
3. **Via DevPortal scripts, it shall be possible to get the last content modification date of the portal contents in designer/editor mode. (prio 1 - urgently required to be able to get information on the status of environments )**
4. Via DevPortal scripts, it shall be possible to get the last published revision information including the publishing date and the content modification mode to check the status of a target environment.
5. Uploads of portal contents via DevPortal scripts shall be indicated on the "Portal overview" page. It shall be visible if the last modification was done within the portal in designer editor mode or from the backend via a content upload.
6. Support version control of content updates in the DevPortal and allow synching with a git repository in Github or Azure DevOps.

## Is your portal managed or self-hosted?

Managed

**Comment by ghost** [comment]
@AnRei123, thank you for opening this issue. We will triage it within the next few business days.

**Comment by mikebudzynski** [comment]
@AnRei123, thanks for requesting those features. Regarding:

> Additionally, it is not possible to restrict the access to the Developer Portal in designer / editor mode separately.

We are working on migration of the portal's administrative interface to Azure Resource Manager APIs (not direct access APIs), which will allow you to set access permissions for portal operations.
